### PR TITLE
CI: migrate pipeline into GH Actions

### PR DESCRIPTION
**Pull request by jportasa**
- [x] Add version print logic into codebase 
- [x] Add docker login for pull rate limitations 
- [x] Add snyk step as `needs` for prerelease once it will be ok
- [x] Migrate into Org secrets
- [x] Change upgrade codes for windows (keep old codes)
- [x] Squash commits because Joan Porta did not accept CLA

### CODE REVIEW

File: build/package/windows/nri-386-installer/Product.wxs [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
-    <?define Version = "{IntegrationVersion}" ?>
+    <?define Version = "$(var.IntegrationVersion)" ?>
     <?define ProductId = "*" ?>
-    <?define UpgradeCode = "{73d03306-78f5-4b3b-a444-fa87381ddadc}" ?>
+    <?define UpgradeCode = "{1a921296-1d82-4014-83ad-9d0c2f9c9bef}" ?>
```
Comment (brushknight) [inline build/package/windows/nri-386-installer/Product.wxs:5]: `UpgradeCode` should be the same

File: build/package/windows/nri-amd64-installer/Product.wxs [hunk 2]
```diff
@@ -1,11 +1,11 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
-    <?define Version = "{IntegrationVersion}" ?>
+    <?define Version = "$(var.IntegrationVersion)" ?>
     <?define ProductId = "*" ?>
-    <?define UpgradeCode = "{73d03306-78f5-4b3b-a444-fa87381ddadc}" ?>
+    <?define UpgradeCode = "{f2334d3a-10f1-4a4c-9185-08d197f95b2b}" ?>
```
Comment (brushknight) [inline build/package/windows/nri-amd64-installer/Product.wxs:5]: Same here with `UpgradeCode `

File: build/package/windows/nri-amd64-installer/Product.wxs [hunk 3]
```diff
@@ -49,47 +49,50 @@
 
     <Fragment>
         <DirectoryRef Id="PluginConfigsFolder">
-            <Component Id="CMP_V1_PLUGIN_CONFIGS" Guid="803C978C-B2A1-47C7-BB17-A2DE9E6D3145" KeyPath="yes">
+            <Component Id="CMP_V1_PLUGIN_CONFIGS" Guid="07bbd5f7-913c-46fa-b6c6-65ab2ca5f695" KeyPath="yes">
```
Comment (paologallinaharbur) [inline build/package/windows/nri-amd64-installer/Product.wxs:52]: Should we change all these componend GUID?
Can it cause issues? @davidbrota @brushknight

**Review by roobre — APPROVED** [review]
Looking hella rad, but let's wait until we reach an agreement for component GUIDs.

**Pull request merged**